src/Angor/Shared/Services/CachedIndexerService.cs

```diff
@@ -0,0 +1,152 @@
+using Angor.Shared.Models;
+using Microsoft.Extensions.Caching.Memory;
+
+namespace Angor.Shared.Services
+{
+    public class CachedIndexerService : ICachedIndexerService
+    {
+        private readonly IIndexerService _indexer;
+        private readonly IMemoryCache _cache;
+        private readonly TimeSpan _cacheExpiration = TimeSpan.FromMinutes(5);
+
+        public CachedIndexerService(IIndexerService indexerService, IMemoryCache cache)
```

Comment:
We don't need to cache every method, just the ones that you identify as problematic, for the rest of them just call directly to the MempoolSpaceIndexerApi you have as a property